src/scorer.coffee

```diff
@@ -42,6 +42,22 @@ exports.basenameScore = (string, query, score) ->
   score *= depth * 0.01
   score
 
+exports.substringScore = (string, query, score) ->
+  # Return a near perfect score if the query is a substring match
+  substrIndex = string.indexOf(query)
+  substrIndexi = string.toLowerCase().indexOf(query.toLowerCase())
```

Comment:
It looks like these two are calculated up front but there are cases where this method will return by only checking `substrIndex`.

Could you then move the calculation of `substrIndexi` down past all the return checks for `substrIndex`? So lower-casing the string and searching it again is only done if the first substring match attempt fails.